src/components/Forum/ForumManagement/RenderUserToBan.js

```diff
@@ -0,0 +1,171 @@
+import { useNavigation } from "@react-navigation/native";
+import { useState } from "react";
+import { Text, Platform } from "react-native";
+import styled from "styled-components/native";
+import { firebase } from "../../../services/Firebase/Config";
+import {
+  deleteBannedUsers,
+  addBannedUsers
+} from "../../../services/Forum/HandleBannedUsers";
+
+const RenderUserToBan = ({
+  username,
+  img,
+  id,
+  reason,
+  isBanned,
+  setIsBanned
+}) => {
+  const [isExpanded, setIsExpanded] = useState(false);
+  const [banDetail, setBanDetail] = useState("");
+  const [banReason, setBanReason] = useState(reason);
+  const navigation = useNavigation();
+  const forumId = navigation.getState().routes[1].params.data.id;
+  const currentUID = firebase.auth().currentUser.uid;
+  const isYou = currentUID === id;
+
+  let buttonMessage;
+
+  //Set Up Different Button Message
+  if (isYou) {
+    buttonMessage = "You can't ban yourself";
+  } else if (isBanned) {
+    buttonMessage = "unban";
+  } else if (isExpanded) {
+    buttonMessage = "cancel";
+  } else {
+    buttonMessage = "ban";
+  }
+
+  function handleButtonPress() {
+    if (isYou) return;
+
+    if (isBanned) {
+      //Unban Button
+      deleteBannedUsers(forumId, id).then(() => {
```

Comment:
If we do this, there is a bug where a user is listed as unbanned before we agree to unban. Instead, we can do
```javascript
Warning( async () => {
  await deleteBannedUsers(forumId, id).then(() => {
    setIsBanned(false);
    setBanReason("");
  });
);
```